### ISSUE TITLE
Haste spell does nothing

### ISSUE THREAD

**Issue by grannypron** [grannypron/uaf_levels#89]
Casting Haste in combat has no effect after asking player to choose a target.

**Comment by grannypron** [comment]
This does NOT happen all the time.  It appears to be related with the following log message: "Failed AddTarget() in COMBAT_SPELL_AIM_MANUAL_MENU_DATA", which is related to this code block in COMBAT_SPELL_AIM_MANUAL_MENU_DATA::OnKeyPress:


```
          if (!m_pCaster->AddMapTarget(combatData.m_iCursorX, combatData.m_iCursorY, dir, dirX, dirY))
          {
            WriteDebugString("Failed AddTarget() in COMBAT_SPELL_AIM_MANUAL_MENU_DATA\n");
          };
```

And is because m_pCharacter->targets.m_targets.GetSize() ends up being 0 at the end of COMBATANT::AddMapTarget()

**Comment by grannypron** [comment]
Debugging reveals that the main source of the problem comes down to this conditional:

```
      if (HaveLineOfSight(VisCheckX,VisCheckY, pt.x, pt.y,NULL))
      {
...
          m_pCharacter->targets.m_maptarglist.AddTail(pt);     
        };
      };

```

And the fact that VisCheckX and VisCheckY are 0.   This is because of all of the cases in the switch above for m_targType (examples include NowAreaCircle, LingerAreaCircle, NowAreaSquare, LingerAreaSquare, NowAreaCone, LingerAreaCone, etc) , the one that applies, NowAreaSquare, does not have these two lines in it like all the rest of the switch statements:

```
      VisCheckX = mapx;
      VisCheckY = mapy;

```

Now, this could be because they were missed, or because there is something special about choosing targets when the NowAreaSquare is the target type.  If that is the case, then something will need to be changed to accommodate whatever that consideration is.  But if that is not the case, I don't know why someone would have not just placed the assignment of VisCheckX, VisCheckY before the whole switch.

However, even in the case of LingerAreaSquare, VisCheckX and VisCheckY are set and I can't think of any kind of reason that would be different from NowAreaSquare.

**Comment by grannypron** [comment]
Adding the assignment to VisCheckX/Y in NowAreaSquare case of the switch does fix the problem.

Brought it up to the group at http://ua.reonis.com/index.php?topic=4031.0

**Comment by grannypron** [comment]
I am going to leave this fix in locally and work on some other bugs while monitoring stability before I commit.

**Comment by grannypron** [comment]
Engine fix tracked at https://github.com/grannypron/uaf/issues/44

**Comment by grannypron** [comment]
Closed with upgrade to 5.27
